python/graphstorm/run/gsgnn_mt/mt_infer_gnn.py

```diff
@@ -0,0 +1,275 @@
+"""
+    Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+    Licensed under the Apache License, Version 2.0 (the "License").
+    You may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    GSgnn multi-task inference entry point.
+"""
+import graphstorm as gs
+from graphstorm.config import get_argument_parser
+from graphstorm.config import GSConfig
+from graphstorm.config import (BUILTIN_TASK_NODE_CLASSIFICATION,
+                               BUILTIN_TASK_NODE_REGRESSION,
+                               BUILTIN_TASK_EDGE_CLASSIFICATION,
+                               BUILTIN_TASK_EDGE_REGRESSION,
+                               BUILTIN_TASK_LINK_PREDICTION,
+                               BUILTIN_TASK_RECONSTRUCT_NODE_FEAT)
+from graphstorm.dataloading import GSgnnData
+from graphstorm.dataloading import (GSgnnNodeDataLoader,
+                                    GSgnnEdgeDataLoader,
+                                    GSgnnMultiTaskDataLoader)
+from graphstorm.dataloading import (GSgnnLinkPredictionTestDataLoader,
+                                    GSgnnLinkPredictionJointTestDataLoader,
+                                    GSgnnLinkPredictionPredefinedTestDataLoader)
+from graphstorm.dataloading import BUILTIN_LP_UNIFORM_NEG_SAMPLER
+from graphstorm.dataloading import BUILTIN_LP_JOINT_NEG_SAMPLER
+
+from graphstorm.model.multitask_gnn import GSgnnMultiTaskSharedEncoderModel
+from graphstorm.inference import GSgnnMultiTaskLearningInferrer
+
+from graphstorm.utils import get_device, use_wholegraph
+from graphstorm.utils import get_lm_ntypes
+from graphstorm.eval import GSgnnMultiTaskEvaluator
+
+def create_task_infer_dataloader(task, config, infer_data):
```

Comment:
Should we move this function into the `gsf.py`?

Reply:
This is only used by infer_gnn.py. So I would like to keep it here now.

Reply:
Alright, might move it once we see other functions use it in the future.